### DISCUSSION TITLE
Crate query result in Job killed

### DISCUSSION THREAD

**Discussion by rjbaat** [crate/crate#12949] [General]
### CrateDB version

4.8.2

### CrateDB setup information

4 node cluster on a k8s cluster. Crate nodes have 10GB ram and 5gb heap.

### Observed behavior

On regular interval I get this message in the PHP application that uses Crate to query devices data. 

```
JobKilledException[Job killed. Received data for job=axxxxxx-xxx-xx-xx but there is no job context present. This can happen due to bad network latency or if individual nodes are unresponsive due to high load
```

I am not sure if the crate cluster is to busy to handle the request. Because it doesn't looks like the cluster is very busy. 

Is there something I can do to improve or make sure the error does't occur?

### Steps to Reproduce

It occurs on a select query on the table that holds the last know state of the device.

**Comment by seut** [comment]
@rjbaat Sorry for the late response, your issue slipped through.
Can you please check the logs of all nodes for `error` entries in the same period your application receive that error and share it?
Without any additional information it is almost impossible for us to help you.

CrateDB will create tasks on each node which is participating in a query execution, these tasks are receiving input from other tasks on other nodes. If a task was removed due to an error, a possible late received payload targeted for this task will fail with such an error. Normally this should not happen as we try our best to handle errors gracefully and forward them to all participating task, so they will stop processing and bubbling up the error. But in some situations this isn't always possible and such resulting in the error you've seen.

I will move this issue into a discussion as it looks more like a support question than a bug/feature.
